src/Sleep.js

```diff
@@ -1,26 +1,45 @@
 class Sleep {
-  constructor() {}
-  userSleepData(sleepData, id) {
-    this.userSleep = sleepData.filter(dailySleep => dailySleep.userID === id);
-    return this.userSleep;
+  constructor(sleepSet) {
+    this.sleepSet = sleepSet
   }
-  averageAllTimeSleep() {
-    let allTimeSleep = this.userSleep.reduce((sleep, day) =>{
-      sleep += day.sleepQuality;
-      return sleep 
+  userSleepData(id) {
+    return this.sleepSet.filter(dailySleep => dailySleep.userID === id);
+  }
+  averageAllTimeSleep(id) {
+    let getUserData = this.userSleepData(id)
+    let allTimeSleep = getUserData.reduce((sleep, day) =>{
+      return sleep += day.sleepQuality;
     }, 0)
-    return Number((allTimeSleep / this.userSleep.length).toFixed(1))
+    return Math.round((allTimeSleep / getUserData.length) * 10) / 10
+  }
+  daySleep(dateSelected, id) {
+    return this.sleepSet.find(day => day.date === dateSelected && day.userID === id).hoursSlept;
   }
-  daySleep(dateSelected) {
-    return this.userSleep.find(day => day.date === dateSelected).hoursSlept;
+  weeklySleepProperties(startDate, id, property) {
+    let startingDate = this.sleepSet.find(day => day.date === startDate && day.userID === id);
+    let firstDay = this.sleepSet.indexOf(startingDate);
+    return this.sleepSet.slice(firstDay, firstDay + 7).map(day => day[property])
+  }
+  averageSleepQuality(allQuality, id) {
+    let dataToAverage = id ? allQuality : this.sleepSet;
+    let userHolder = id ? id : null;
+    let average = dataToAverage.reduce((quality, user) => {
+      return quality += userHolder ? user : user.sleepQuality;
+    }, 0)
+    return Math.round(average / dataToAverage.length * 10) / 10;
   }
-  dailySleepPerGivenWeek(startDate) {         
-    let startingDate = this.userSleep.find(day => day.date === startDate);
-    let firstDay = this.userSleep.indexOf(startingDate);
-    return this.userSleep.slice(firstDay, firstDay + 7).map(day => day.hoursSlept)
+  sleepQualityAboveThree(date) {
+    let qualityAboveThree = [];
+    let uniqueIds = [];
+    this.sleepSet.forEach(user => !uniqueIds.includes(user.userID) ? uniqueIds.push(user.userID) : null)
+    uniqueIds.forEach(id => {
+      let allQuality = this.weeklySleepProperties(date, id, 'sleepQuality');
+      let averageQuality = this.averageSleepQuality(allQuality, id);
+      averageQuality > 3 ? qualityAboveThree.push(id) : null;
```

Comment:
This method is a little bit more slimmed down, but is still a bit hard to read.

* `qualityAboveThree` will be the array we return that will contain the passing IDS

* `uniqueIds` will filter through the sleep data and pull only unique Ids

our forEach will now only be run for the amount of IDs there are which is way better than what we previously had.

*ForEach ID* 

*  `allQuality` variable will hold the weekly `sleepQuality` values for the specific ID we are iterating on
*  `averageQuality` variable will hold the return value of our `averageSleepQuality` 
*.  if averageQuality is above 3 we will push it to our array, if else, we do nothing.

We will then return our array of all >3 IDs, can probably push more than just ID if we deem that necessary

Reply:
I agree I think we still might be able to do more here to slim this down more and make it more readable